spec/tapioca/compilers/dsl/active_record_columns_spec.rb

```diff
@@ -0,0 +1,336 @@
+
+          sig { void }
+          def body_will_change!; end
+      RUBY
+
+      output = rbi_for(files)
+      expect(output).to(include(expected))
+
+      expected = indented(<<~RUBY, 2)
+        sig { void }
+        def restore_body!; end
+      RUBY
+      expect(output).to(include(expected))
+
+      expected = indented(<<~RUBY, 2)
+        sig { returns([T.nilable(::String), T.nilable(::String)]) }
+        def saved_change_to_body; end
+
+        sig { returns(T::Boolean) }
+        def saved_change_to_body?; end
+      RUBY
+      expect(output).to(include(expected))
+    end
+
+    it("generates RBI file for custom attributes without strong type generation") do
+      files = {
+        "file.rb" => <<~RUBY,
+          module StrongTypeGeneration
+          end
+
+          class Post < ActiveRecord::Base
+            # StrongTypeGeneration is not extended
+          end
+        RUBY
+
+        "schema.rb" => <<~RUBY,
+          ActiveRecord::Migration.suppress_messages do
+            ActiveRecord::Schema.define do
+              create_table :posts do |t|
+                t.string :body
+              end
+            end
+          end
+        RUBY
+      }
+
+      expected = <<~RUBY
+        module Post::GeneratedAttributeMethods
+          sig { returns(T.untyped) }
+          def body; end
+
+          sig { params(value: T.untyped).returns(T.untyped) }
+          def body=(value); end
+
+          sig { returns(T::Boolean) }
+          def body?; end
+
+          sig { returns(T.untyped) }
+          def body_before_last_save; end
+
+          sig { returns(T.untyped) }
+          def body_before_type_cast; end
+
+          sig { returns(T::Boolean) }
+          def body_came_from_user?; end
+
+          sig { returns([T.untyped, T.untyped]) }
+          def body_change; end
+
+          sig { returns([T.untyped, T.untyped]) }
+          def body_change_to_be_saved; end
+
+          sig { returns(T::Boolean) }
+          def body_changed?; end
+
+          sig { returns(T.untyped) }
+          def body_in_database; end
+
+          sig { returns([T.untyped, T.untyped]) }
+          def body_previous_change; end
+
+          sig { returns(T::Boolean) }
+          def body_previously_changed?; end
+
+          sig { returns(T.untyped) }
+          def body_previously_was; end
+
+          sig { returns(T.untyped) }
+          def body_was; end
+
+          sig { void }
+          def body_will_change!; end
+      RUBY
+
+      output = rbi_for(files)
+      expect(output).to(include(expected))
+
+      expected = indented(<<~RUBY, 2)
+        sig { void }
+        def restore_body!; end
+      RUBY
+      expect(output).to(include(expected))
+
+      expected = indented(<<~RUBY, 2)
+        sig { returns([T.untyped, T.untyped]) }
+        def saved_change_to_body; end
+
+        sig { returns(T::Boolean) }
+        def saved_change_to_body?; end
+      RUBY
+      expect(output).to(include(expected))
+    end
```

Comment:
Multiple `expect` calls were used here because the output is not continuous. This is what I could come up with but open to suggestions.